folium/features.py

```diff
@@ -328,10 +343,36 @@ def style_function(x):
                 return {}
         self.style_function = style_function
 
+        self.popup_attribute = None
+        if isinstance(popup_function, FunctionType):
+            self.popup_attribute = "_popupContent"
+            try:
+                for feature in self.data["features"]:
+                    feature["properties"]["_popupContent"] = popup_function(feature)
+            except TypeError:
+                raise ValueError("Can't use popup_function on file path")
+        elif isinstance(popup_function, text_type):
+            self.popup_attribute = popup_function
+        elif isinstance(popup_function, binary_type):
+            self.popup_attribute = text_type(popup_function, 'utf8')
+
```

Comment:
This is really great ! Thanks.

I would prefer to move this paragraph (346-358) into `style_data` and simply write `self.popup_function = popup_function` here. That way, the following code will also work:

```python
g = GeoJson(data)
g.popup_function = some_function
```